### PR TITLE
Export privs

### PR DESCRIPTION
**Pull request by benrudolph**
@emord @nickpell @NoahCarnahan @calellowitz made a management command to grandfather in all roles. i added it to a migration which means that on a new setup, it'll get run on again which i think is OK. tested out locally with my `brudolph@dimagi.com_privileges` role and it seemed to work. i think worth doing a dry run on prod before hand to ensure that we don't do anything accidental. can be read 🐟 or 🏠

### CODE REVIEW

**Comment by benrudolph** [comment]
http://manage.dimagi.com/default.asp?237494

File: corehq/apps/accounting/utils.py [hunk 1]
```diff
@@ -278,3 +278,41 @@ def get_default_domain_url(domain):
...
+def ensure_grant(grantee_slug, priv_slug, dry_run=False, verbose=False):
+    """
+    Adds a parameterless grant between grantee and priv, looked up by slug.
+    """
+
+    if dry_run:
+        grants = Grant.objects.filter(from_role__slug=grantee_slug,
+                                      to_role__slug=priv_slug)
+        if not grants:
+            logger.info('[DRY RUN] Granting privilege: %s => %s', grantee_slug, priv_slug)
```
Comment (emord) [inline corehq/apps/accounting/utils.py]: Noticed you just copied this, so no action required, but do you think it's worthwhile to add "privilege already granted" to the dry run? might be useful for auditing if we accidentally granted a privilege to a role that shouldn't be there

File: corehq/privileges.py [hunk 2]
```diff
@@ -87,7 +87,7 @@
     DATA_CLEANUP,
     TEMPLATED_INTENTS,
     CUSTOM_INTENTS,
-    # BUILD_PROFILES,
+    BUILD_PROFILES,
```
Comment (emord) [inline corehq/privileges.py]: What's the reason for adding this back in?

**Review by emord — COMMENTED** [review]
Looks good to me. Nothing blocking assuming your tests and dry runs go well

File: corehq/privileges.py (hunk 2 above)
Comment (benrudolph) [inline corehq/privileges.py]: the assumption i make in my code is that all privileges are added to max privileges (roles - max_privileges == all plans). it wouldn't make sense to add daily saved exports to the build profile privilege. also talked to cal and this is supposed to be in max privs

File: corehq/apps/accounting/utils.py (hunk 1 above)
Comment (benrudolph) [inline corehq/apps/accounting/utils.py]: yes that would be useful, adding

**Comment by benrudolph** [comment]
here's the output from a run on staging:
```
2016-09-16 16:43:05,828 INFO [DRY RUN] Granting privilege: npellegrino@dimagi.com_privileges => excel_dashboard
2016-09-16 16:43:05,864 INFO [DRY RUN] Privilege already granted: standard_plan_v0 => excel_dashboard
2016-09-16 16:43:05,867 INFO [DRY RUN] Granting privilege: jjackson@dimagi.com_privileges => excel_dashboard
2016-09-16 16:43:05,870 INFO [DRY RUN] Privilege already granted: advanced_plan_v0 => excel_dashboard
2016-09-16 16:43:05,873 INFO [DRY RUN] Granting privilege: sshah@dimagi.com_privileges => excel_dashboard
2016-09-16 16:43:05,876 INFO [DRY RUN] Granting privilege: brudolph@dimagi.com_privileges => excel_dashboard
2016-09-16 16:43:05,879 INFO [DRY RUN] Granting privilege: ksharpey@dimagi.com_privileges => excel_dashboard
2016-09-16 16:43:05,882 INFO [DRY RUN] Granting privilege: esoergel@dimagi.com_privileges => excel_dashboard
2016-09-16 16:43:05,885 INFO [DRY RUN] Granting privilege: ncarnahan@dimagi.com_privileges => excel_dashboard
2016-09-16 16:43:05,888 INFO [DRY RUN] Granting privilege: cellowitz@dimagi.com_privileges => excel_dashboard
2016-09-16 16:43:05,891 INFO [DRY RUN] Granting privilege: accounting_admin => excel_dashboard
2016-09-16 16:43:05,894 INFO [DRY RUN] Granting privilege: biyeun@dimagi.com_privileges => excel_dashboard
2016-09-16 16:43:05,897 INFO [DRY RUN] Granting privilege: jkass@dimagi.com_privileges => excel_dashboard
2016-09-16 16:43:05,900 INFO [DRY RUN] Granting privilege: dimagi_ops => excel_dashboard
2016-09-16 16:43:05,903 INFO [DRY RUN] Privilege already granted: pro_plan_v0 => excel_dashboard
2016-09-16 16:43:05,906 INFO [DRY RUN] Granting privilege: Advanced_Report_Builder_Trial => excel_dashboard
2016-09-16 16:43:05,909 INFO [DRY RUN] Granting privilege: nclark@dimagi.com_privileges => excel_dashboard
2016-09-16 16:43:05,912 INFO [DRY RUN] Granting privilege: ssnyder@dimagi.com_privileges => excel_dashboard
2016-09-16 16:43:05,915 INFO [DRY RUN] Privilege already granted: community_plan_v0 => excel_dashboard
2016-09-16 16:43:05,918 INFO [DRY RUN] Granting privilege: ksmith@dimagi.com_privileges => excel_dashboard
2016-09-16 16:43:05,921 INFO [DRY RUN] Granting privilege: cwhite@dimagi.com_privileges => excel_dashboard
2016-09-16 16:43:05,924 INFO [DRY RUN] Granting privilege: czue@dimagi.com_privileges => excel_dashboard
2016-09-16 16:43:05,927 INFO [DRY RUN] Granting privilege: frener@dimagi.com_privileges => excel_dashboard
2016-09-16 16:43:05,930 INFO [DRY RUN] Granting privilege: cpowers@dimagi.com_privileges => excel_dashboard
2016-09-16 16:43:05,933 INFO [DRY RUN] Granting privilege: asagoff@dimagi.com_privileges => excel_dashboard
2016-09-16 16:43:05,936 INFO [DRY RUN] Granting privilege: kswamy@dimagi.com_privileges => excel_dashboard
2016-09-16 16:43:05,939 INFO [DRY RUN] Privilege already granted: enterprise_plan_v0 => excel_dashboard
2016-09-16 16:43:05,942 INFO [DRY RUN] Granting privilege: bulk_form_management => excel_dashboard
2016-09-16 16:43:05,945 INFO [DRY RUN] Granting privilege: droberts@dimagi.com_privileges => excel_dashboard
2016-09-16 16:43:05,948 INFO [DRY RUN] Granting privilege: nhaduch@dimagi.com_privileges => excel_dashboard
```
everything makes sense except for the bulk_form_management which i think is just an oddity on staging since it's not specified anywhere in the code as far as i can tell

File: corehq/privileges.py (hunk 2 above)
Comment (emord) [inline corehq/privileges.py]: That sounds reasonable

**Comment by emord** [comment]
This is good to go from me. Will wait on merging in case you want to setup_release and dry run on prod before migrating

File: corehq/apps/hqadmin/management/commands/cchq_prbac_grandfather_privs.py [hunk 3]
```diff
@@ -43,8 +43,8 @@ def handle(self, *privs, **kwargs):
         verbose = kwargs.get('verbose')
         noinput = kwargs.get('noinput')
         all_plan_slugs = (
-            set(map(lambda role: role.slug, Role.objects.all())) -
-            set(MAX_PRIVILEGES) -
+            set(map(lambda s: s.role.slug, SoftwarePlanVersion.objects.all())) -
```
Comment (benrudolph) [inline corehq/apps/hqadmin/management/commands/cchq_prbac_grandfather_privs.py]: @emord @nickpell did some testing with this and this is much more precise

**Comment by emord** [comment]
You should also be able to revert this https://github.com/dimagi/commcare-hq/pull/13257/files right?

**Comment by benrudolph** [comment]
yep (will do it after this)

**Comment by emord** [comment]
cool. This is good to go from my side. the last test is still running so not merging yet (doesn't show since last commit is [ci-skip]), but I don't think that this will be tested anyways

**Comment by benrudolph** [comment]
ok merging, dry ran on production and looks good

**Comment by NoahCarnahan** [comment]
Thanks Ben!

**Pull request merged**